chainermn/communicators/pure_nccl_communicator.py

```diff
@@ -157,6 +144,38 @@ def _unpack_params_from_buffer(self, params,
                 params, 'grad', self.gpu_buffer_a,
                 allreduce_grad_dtype, stream)
 
+    def multinode_mean_nccl(self, gpu_buffer_a, gpu_buffer_b,
+                            n_elems, dtype, stream=None):
+        # Performs allreduce and division by size, i.e. mean.
+        # gpu_buffer_a = Sigma(gpu_buffer_a, all-procs) / self.size
+        # b is just used as buffer
+        if chainer.is_debug():
+            stream.synchronize()
+            array_a = gpu_buffer_a.array(n_elems, dtype=dtype)
+            array_b = gpu_buffer_b.array(n_elems, dtype=dtype)
+            self.check_ready_to_allreduce(array_a, array_b)
+
+        if stream is None:
+            stream = chainer.cuda.Stream.null):
+        self._init_comms()
+        type_id = _communication_utility._get_nccl_type_id(dtype),
+        self.nccl_comm.allReduce(gpu_buffer_a.ptr(),
+                                 gpu_buffer_b.ptr(), n_elems,
+                                 type_id, nccl.NCCL_SUM, stream.ptr)
+        if self.div_by_size is None:
+            self.div_by_size = chainer.cuda.cupy.ElementwiseKernel(
+                '{} x'.format(dtype.name),
+                '{} y'.format(dtype.name),
+                'y = x*(1.0/{})'.format(self.size), 'div_by_size')
+        self.div_by_size(
+            gpu_buffer_b.array(n_elems, dtype=dtype),
+            gpu_buffer_a.array(n_elems, dtype=dtype),
+            stream=stream)
+
+        if chainer.is_debug():
+            stream.synchronize()
+            self.ensure_all_finite(gpu_buffer_b.array(n_elems, dtype=dtype))
```

Comment:
The `gpu_buffer_b` is tmp buffer in `multinode_mean_nccl`. 
How about checking `gpu_buffer_a` ?

Reply:
Good catch! thanks